src/api/initialDataTypes.js

```diff
@@ -100,7 +100,15 @@ export type InitialDataRealm = $ReadOnly<{|
   realm_invite_required: boolean,
   realm_is_zephyr_mirror_realm: boolean,
   realm_mandatory_topics: boolean,
-  realm_message_content_delete_limit_seconds: number,
+
+  realm_message_content_delete_limit_seconds: | (number | null)
+    // TODO(server-5.0): We can switch to just `number | null`. In 5.0
```

Comment:
Ah, right! So I do understand that this is the same type as `number | null`. 🙂

I just thought it might be helpful for readability, along with some comments, to describe the two different representations we'll get from the server (before 5.0, and at 5.0+). With the `number | null` branch really wanting to be `number-except-0 | null`, I guess.

Reply:
Cool. Yeah, I think it's clearest to let the type here be in the more simplified/normalized form, and cover the specifics of the variation just in the text.